tests/test_lightblue.py

```diff
@@ -3099,9 +3099,34 @@ def test_filter_out_already_fixed_published_images(mock_dig, mock_gfpi, mock_exi
     )
 
 
+@pytest.mark.usefixtures("pyxis_graphql_schema")
+@patch('freshmaker.pyxis_gql.Client')
 @patch('os.path.exists', return_value=True)
-@patch('freshmaker.lightblue.LightBlue.find_container_images')
-def test_get_fixed_published_image(mock_fci, mock_exists):
+@patch('freshmaker.pyxis_gql.PyxisGQL.find_published_images_by_name_version_content_sets')
+@patch('freshmaker.pyxis_gql.PyxisGQL.find_images_by_nvr')
+@patch('freshmaker.lightblue.LightBlue.postprocess_images')
```

Comment:
@qixiang I checked it, and in most places I was able to omit mocking the `postprocess_images` method, but in this specific place (line 3107) I had to leave it mocked because otherwise, here https://github.com/redhat-exd-rebuilds/freshmaker/blob/master/freshmaker/lightblue.py#L1939 the `resolve` was causing an error.

Reply:
That error is not caused by `postprocess_images`, it'd be better to just mock `freshmaker.lightblue.ContainerImage.resolve` and remove the related mocking data for `postprocess_images`.

Reply:
```
@pytest.mark.usefixtures("pyxis_graphql_schema")
@patch('freshmaker.pyxis_gql.Client')
@patch('os.path.exists', return_value=True)
@patch('freshmaker.pyxis_gql.PyxisGQL.find_images_by_name_version')
@patch('freshmaker.pyxis_gql.PyxisGQL.find_images_by_nvr')
@patch('freshmaker.lightblue.ContainerImage.resolve')
def test_get_fixed_published_image(resolve, find_images_by_nvr, published_images, mock_exists, gql_client):
    other_rhel7_image_pyxis = {
        "brew": {"build": "rhel-server-container-7.9-188"},
        "content_sets": ["rhel-7-server-rpms"],
        "repositories": [{"repository": "repo"}],
        "edges": {
            "rpm_manifest": {
                "data": {
                    "image_id": '57ea8dc69c624c035f96f990',
                    "rpms": [
                        {
                            "name": "bash",
                            "nvra": "bash-4.2.46-34.el7.x86_64",
                            "srpm_name": "bash",
                            "srpm_nevra": "bash-0:4.2.46-34.el7.src",
                            "version": "4.2.46",
                        }
                    ]
                }
            }
        }
    }
    latest_rhel7_image_pyxis = {
        "brew": {"build": "rhel-server-container-7.9-189"},
        "content_sets": ["rhel-7-server-rpms"],
        "repositories": [{"repository": "repo"}],
        "edges": {
            "rpm_manifest": {
                "data": {
                    "image_id": '57ea8dc69c624c035f96f990',
                    "rpms": [
                        {
                            "name": "bash",
                            "nvra": "bash-4.2.46-34.el7.x86_64",
                            "srpm_name": "bash",
                            "srpm_nevra": "bash-0:4.2.46-34.el7.src",
                            "version": "4.2.46",
                        }
                    ]
                }
            }
        }
    }

    published_images.return_value = [other_rhel7_image_pyxis, latest_rhel7_image_pyxis]
    find_images_by_nvr.return_value = [latest_rhel7_image_pyxis]
    image = Mock(nvr="rhel-server-container-7.9-185")
    image.get_registry_repositories.return_value = [{"repository": "repo"}]
    image_group = ImageGroup(image, Mock())
    rpm_nvrs = ["bash-4.2.46-34.el7"]
    content_sets = ["rhel-7-server-rpms"]
    lb = LightBlue("lb.domain.local", "/path/to/cert", "/path/to/key")

    image = lb.get_fixed_published_image(
        "rhel-server-container", "7.9", image_group, rpm_nvrs, content_sets
    )

    assert image["brew"]["build"] == "rhel-server-container-7.9-189"

    latest_rhel7_image_pyxis = {
        "brew": {"build": "rhel-server-container-7.9-189"},
        "content_sets": ["rhel-7-server-rpms"],
        "repositories": [{"repository": "repo2"}],
        "edges": {
            "rpm_manifest": {
                "data": {
                    "image_id": '57ea8dc69c624c035f96f990',
                    "rpms": [
                        {
                            "name": "bash",
                            "nvra": "bash-4.2.46-34.el7.x86_64",
                            "srpm_name": "bash",
                            "srpm_nevra": "bash-0:4.2.46-34.el7.src",
                            "version": "4.2.46",
                        }
                    ]
                }
            }
        }
    }

    published_images.return_value = [other_rhel7_image_pyxis, latest_rhel7_image_pyxis]
    find_images_by_nvr.return_value = [latest_rhel7_image_pyxis]

    image_2 = Mock(nvr="rhel-server-container-7.9-185")
    image_2.get_registry_repositories.return_value = [{"repository": "repo"}]
    image_group_2 = ImageGroup(image_2, Mock())
    image_2 = lb.get_fixed_published_image(
        "rhel-server-container", "7.9", image_group_2, rpm_nvrs, content_sets
    )

    assert image_2["brew"]["build"] == "rhel-server-container-7.9-189"
```